### ISSUE TITLE
Add a new docker-compose file which uses the pre-built images for ATLAS and WebAPI

### ISSUE THREAD

**Issue by chgl** [OHDSI/Broadsea#39]
Follow-up from https://github.com/OHDSI/Broadsea-WebTools/issues/14#issuecomment-1302878315

It would be nice to use the pre-built images for WebAPI and ATLAS when deploying via docker-compose. This might additionally reduce the maintenance burden of keeping the [Broadsea-WebTools](https://github.com/OHDSI/Broadsea-WebTools) image up-to-date.

An example docker-compose.yaml may look as follows:

```yaml
services:
  traefik:
    image: docker.io/library/traefik:v2.9.4@sha256:4c388f477f5b7aa48420694d90db0ecb96b4ecede60b9b7589c87858abf87b60
    restart: unless-stopped
    ipc: none
    security_opt:
      - "no-new-privileges:true"
    cap_drop:
      - ALL
    read_only: true
    command:
      - "--providers.docker=true"
      - "--providers.docker.exposedbydefault=false"
      - "--entrypoints.web.address=:80"
    ports:
      - "127.0.0.1:80:80"
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock:ro"

  ohdsi-db:
    image: docker.io/library/postgres:15.0@sha256:bab8d7be6466e029f7fa1e69ff6aa0082704db330572638fd01f2791824774d8
    restart: unless-stopped
    ipc: private
    security_opt:
      - "no-new-privileges:true"
    shm_size: 1024m
    privileged: false
    environment:
      POSTGRES_PASSWORD: postgres
      POSTGRES_DB: ohdsi

  ohdsi-webapi:
    image: docker.io/ohdsi/webapi:2.12.0@sha256:4d951a05245cedad4d666874fb71f182bd2140aab5ece4f1621d028cf10d0823
    restart: unless-stopped
    ipc: none
    security_opt:
      - "no-new-privileges:true"
    cap_drop:
      - ALL
    read_only: true
    tmpfs:
      - /tmp
    privileged: false
    environment:
      DATASOURCE_DRIVERCLASSNAME: org.postgresql.Driver
      DATASOURCE_URL: jdbc:postgresql://ohdsi-db:5432/ohdsi
      DATASOURCE_USERNAME: postgres
      DATASOURCE_PASSWORD: postgres
      DATASOURCE_OHDSI_SCHEMA: ohdsi
      SPRING_JPA_PROPERTIES_HIBERNATE_DIALECT: org.hibernate.dialect.PostgreSQLDialect
      SPRING_JPA_PROPERTIES_HIBERNATE_DEFAULT_SCHEMA: ohdsi
      SPRING_BATCH_REPOSITORY_TABLEPREFIX: ohdsi.BATCH_
      FLYWAY_DATASOURCE_DRIVERCLASSNAME: org.postgresql.Driver
      FLYWAY_DATASOURCE_URL: jdbc:postgresql://ohdsi-db:5432/ohdsi
      FLYWAY_DATASOURCE_USERNAME: postgres
      FLYWAY_DATASOURCE_PASSWORD: postgres
      FLYWAY_LOCATIONS: classpath:db/migration/postgresql
      FLYWAY_PLACEHOLDERS_OHDSISCHEMA: ohdsi
      FLYWAY_SCHEMAS: ohdsi
      SECURITY_CORS_ENABLED: "true"
      SECURITY_ORIGIN: "http://broadsea-webtools.127.0.0.1.nip.io"
    depends_on:
      - ohdsi-db
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.ohdsi-webapi.rule=Host(`broadsea-webtools.127.0.0.1.nip.io`) && PathPrefix(`/WebAPI`)"
      - "traefik.http.routers.ohdsi-webapi.entrypoints=web"

  ohdsi-atlas:
    image: docker.io/ohdsi/atlas:2.12.0@sha256:96431e12cba41643117a904f14c3c6d50a0af63a9a293ed61cd56d5bcffb723a
    restart: unless-stopped
    ipc: none
    security_opt:
      - "no-new-privileges:true"
    cap_drop:
      - ALL
    privileged: false
    environment:
      WEBAPI_URL: "http://broadsea-webtools.127.0.0.1.nip.io/WebAPI/"
    depends_on:
      - ohdsi-webapi
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.ohdsi-atlas.rule=Host(`broadsea-webtools.127.0.0.1.nip.io`) && PathPrefix(`/atlas`)"
      - "traefik.http.routers.ohdsi-atlas.entrypoints=web"
```

This includes [traefik](https://traefik.io/) as a reverse proxy in front of both ATLAS and the WebAPI, exposed on the same hostname but distinguished by path ("/atlas" vs "/WebAPI"). Traefik is generally a bit easier to configure when used with containers as opposed to nginx and its configuration files. Instead of using a reverse proxy, for local installations its also possible to just expose both services on different ports bound to localhost. The example above uses https://nip.io/ to provide "nicer" URLs instead of localhost.

Since the use of traefik is entirely optional we could provide an option to disable it by using profiles (https://docs.docker.com/compose/profiles/) or by using multiple docker-compose files and merging them (https://docs.docker.com/compose/extends/).

I'd be happy to create a PR to include the above compose deployment. A few ideas:

1. I'm wondering if it would make sense to remove the oracle, impala, sqlserver samples which are nearly identical and instead provide just one canonical postgresql-based compose file that can be used as-is. 
2. To keep the container image tags (and digests) up-to-date automatically, it might be helpful to add https://github.com/renovatebot/renovate to the repository, whenever a new version of any of the images used in the docker-compose file is released, renovate will create a PR to update it. This could be even more automated by auto-merging these PRs and creating releases (this is what I'm doing over at https://github.com/chgl/kube-powertools to avoid manual work).
3. I'd probably go for the option to provide traefik as an optional dependency in its own docker-compose.yaml which can be merged with the default compose file.

cc @t-abdul-basser

**Comment by t-abdul-basser** [comment]
1. This makes sense given that the latest version of WebAPI only supports Postgres for the OHDSI/WebAPI application database.
2. Make sense. Automated PR generation is preferable for now.
3. Makes sense.
Thoughts @leeevans?

**Comment by leeevans** [comment]
@chgl - I've now implemented a number of your enhancements in Broadsea 2.0, here in this repo - thanks!

- Use the auto built OHDSI/atlas and OHDSI/WebAPI containers - retire Broadsea-webtools
- Use Traefik reverse proxy

I did not implement  URLs of the form "http://broadsea-webtools.127.0.0.1.nip.io" because I did not want to introduce a dependency on a third-party solution which would also require internet access on the host running Broadsea.

Other new features I have implemented in Broadsea 2.0:

- Use of a new postgres database container OHDSI/broadsea-atlasdb. It is a PostgreSQL database I created which includes a complete demo CDM (Eunomia dataset) and also can be used to connect to the user's own large CDM database server by just configuring the source/source_daimon table data file.
- Broadsea website content: This is a simple HTML page that gives the user some info on each Broadsea application and a link to open it in a new browser page. Accessible through http:127.0.0.1/broadsea.  It can be expanded/customized by the user. I plan to add additional Broadsea applications in the future, including ARES and Perseus.
- I added Hades (inside RStudio) as an additional Broadsea application accessible through Traefik. I wasn't able to get it to work with a path of http:/127.0.0.1/hades so I just ended up redirecting to port 8787 (Rstudio) as a default Traefik path. If you could make the hades path work in Traefik, please submit a PR.

**Comment by leeevans** [comment]
@t-abdul-basser thanks for sharing your forked repo commit as an example.  I think the above changes incorporate that so I am now closing this issue as implemented.
